src/main/java/org/broadinstitute/hellbender/utils/codecs/GENCODE/GencodeGtfCodec.java

```diff
@@ -0,0 +1,376 @@
+
+            // Ensure the file is at least trivially well-formed:
+            if (splitLine.length != NUM_COLUMNS) {
+                throw new UserException.MalformedFile("Found an invalid number of columns in the given GENCODE file on line "
+                        + currentLineNum + " - Given: " + splitLine.length + " Expected: " + NUM_COLUMNS + " : " + line);
+            }
+
+            // We need to key off the feature type to collapse our accumulated records:
+            final String featureType = splitLine[2];
+
+            // Create a baseline feature to add into our data:
+            GencodeGtfFeature feature = GencodeGtfFeature.create(splitLine);
+
+            // Make sure we keep track of the line number for if and when we need to write the file back out:
+            feature.setFeatureOrderNumber(currentLineNum);
+
+            // Once we see another gene, we take all accumulated records and combine them into a
+            // GencodeGtfFeature.
+            if ((gene != null) && (transcript != null) && (featureType.equals("gene") || featureType.equals("transcript") )) {
+
+                aggregateRecordsIntoGeneFeature(gene, transcript, exonStore, leafFeatureStore);
+
+                if ( featureType.equals("gene") ) {
+                    decodedFeature = gene;
+                    break;
+                }
+                else if ( featureType.equals("transcript") ) {
+                    transcript = (GencodeGtfTranscriptFeature) feature;
+                    ++currentLineNum;
+                }
+            }
+            else {
+                // We have not reached the end of this set of gene / transcript records.
+                // We must cache these records together so we can create a meaningful data hierarchy from them all.
+                // Records are stored in their Feature form, not string form.
+
+                // Add the feature to the correct storage unit for easy assembly later:
+                switch (featureType) {
+                    case "gene":
+                        gene = (GencodeGtfGeneFeature)feature;
+                        break;
+                    case "transcript":
+                        transcript = (GencodeGtfTranscriptFeature)feature;
+                        break;
+                    case "exon":
+                        exonStore.add((GencodeGtfExonFeature)feature);
+                        break;
+                    default:
+                        leafFeatureStore.add(feature);
+                        break;
+                }
+
+                ++currentLineNum;
+            }
+
+            // Increment our iterator here so we don't accidentally miss any features from the following gene
+            lineIterator.next();
+        }
+
+        // Do we have some records leftover that we need to aggregate into a feature:
+        if ( (gene != null) &&
+                (exonStore.size() != 0) || (leafFeatureStore.size() != 0) ) {
```

Comment:
Yes, exactly.  I'll add a comment.

Yes - I just devised a case where if the transcript is the last line of the file and there are no other features below it, the transcript will not be added.  It's not clear to me that this is a valid GENCODE GTF file, but I will add in code to handle this case.